_posts/2020-12-04-release-1.12.0.md

```diff
@@ -0,0 +1,313 @@
+
+<div class="alert alert-danger small" markdown="1">
+<b>Attention:</b> The unified source implementations will be completely separate connectors that are not snapshot-compatible with their legacy counterparts.
+</div>
+
+**Pipelined Region Scheduling ([FLIP-119](https://cwiki.apache.org/confluence/display/FLINK/FLIP-119+Pipelined+Region+Scheduling#FLIP119PipelinedRegionScheduling-BulkSlotAllocation))**
+
+Flink’s scheduler has been largely designed to address batch and streaming workloads separately. This release introduces a **unified** scheduling strategy that identifies sets of tasks connected via blocking data exchanges to break down the execution graph into _pipelined regions_. This allows to schedule each region only when there’s data to perform work and only deploy it once all the required resources are available; as well as to restart failed regions independently. In particular for batch jobs, the new strategy leads to more efficient resource utilization and eliminates deadlocks.
+
+**Support for Sort-Merge Blocking Shuffles ([FLIP-148](https://cwiki.apache.org/confluence/display/FLINK/FLIP-148%3A+Introduce+Sort-Merge+Based+Blocking+Shuffle+to+Flink))**
+
+To improve the stability, performance and resource utilization of large-scale batch jobs, the community introduced sort-merge blocking shuffle as an alternative to the hash-based implementation that Flink already used. This approach can reduce shuffle time [significantly](https://www.mail-archive.com/dev@flink.apache.org/msg42472.html). Further optimizations will be implemented in upcoming releases ([FLINK-19614](https://issues.apache.org/jira/browse/FLINK-19614)).
+
+<div class="alert alert-danger small" markdown="1">
+<b>Attention:</b> This feature is experimental and not enabled by default. To enable sort-merge shuffles, you can configure a reasonable minimum parallelism threshold in the <a href="https://ci.apache.org/projects/flink/flink-docs-master/ops/config.html#taskmanager-network-sort-shuffle-min-parallelism">TaskManager network configuration options</a>.
+</div>
+
+**Improvements to the Flink WebUI ([FLIP-75](https://cwiki.apache.org/confluence/display/FLINK/FLIP-75%3A+Flink+Web+UI+Improvement+Proposal))**
+
+As a continuation of the series of improvements to the Flink WebUI kicked off in the last release, the community worked on exposing JobManager's memory-related metrics and configuration parameters on the WebUI ([FLIP-104](https://cwiki.apache.org/confluence/display/FLINK/FLIP-104%3A+Add+More+Metrics+to+Jobmanager)). The TaskManager's metrics page has also been updated to reflect the [changes to the TaskManager memory model](https://flink.apache.org/news/2020/04/21/memory-management-improvements-flink-1.10.html) introduced in Flink 1.10 ([FLIP-102](https://cwiki.apache.org/confluence/display/FLINK/FLIP-102%3A+Add+More+Metrics+to+TaskManager)), adding new metrics for Managed Memory, Network Memory and Metaspace.
```

Comment:
Shall we add a screen shot here? Web UI changes are one of the few release items where we can show something nice ;)

Reply:
The only reason I didn't add a screenshot was because the blogpost is already very extensive. Wouldn't be against having it if you think it's fine and benefits the reader.